pkg/controller/controller_instance.go

```diff
@@ -204,13 +204,15 @@ func (c *controller) reconcileServiceInstanceDelete(instance *v1alpha1.ServiceIn
 		return nil
 	}
 
-	// If there is no op in progress, and the instance was never provisioned,
-	// we can just delete. this can happen if the service class name
-	// referenced never existed.
-	//
-	// TODO: the above logic changes slightly once we handle orphan
-	// mitigation.
-	if !instance.Status.AsyncOpInProgress && instance.Status.ReconciledGeneration == 0 {
+	// If there is no op in progress, and the instance either was never
+	// provisioned or was already deprovisioned due to orphan mitigation,
+	// we can just clear the finalizer and delete. One possible  scenario
+	// is if  the service class name referenced never existed.
+	if !instance.Status.AsyncOpInProgress &&
+		!instance.Status.OrphanMitigationInProgress &&
+		(isServiceInstanceFailed(instance) || instance.Status.ReconciledGeneration == 0) {
+
+		glog.V(5).Infof("Clearing catalog finalizer from ServiceInstance %v/%v", instance.Namespace, instance.Name)
 		clone, err := api.Scheme.DeepCopy(instance)
 		if err != nil {
 			return err
```

Comment:
@kibbles-n-bytes 

Some of the comments in this method now need to be reworked because we could be either handling a delete OR doing orphan mitigation.  For example:

```go
// All updates not having a DeletingTimestamp will have been handled above
// and returned early. If we reach this point, we're dealing with an update
// that's actually a soft delete-- i.e. we have some finalization to do.
```

...that isn't quite true anymore.

I believe the implementation of this is correct, but we need a follow-up to ensure the comments are accurate.